operator_toolkit/check_auto.py

```diff
@@ -1,118 +1,88 @@
-
-
-def is_circuit_control_in_auto(site):
-    """Checks if the circuits of circuit-control are on AUTO status.
-
-    :param str site: site id.
-    :return: True if circuits are in auto, False otherwise.
-    :rtype: bool
-    """
-    url = Operator.generate_url(ServiceType.CIRCUIT_CONTROL, site, path="/report")
-    response = Operator.request(url)
-    if response["status"] != HTTPStatus.OK:
-        return {"error": "Could not get circuits information"}
-    report: dict = response["content"]
-    circuits_status = [value["status"] if "status" in value else value
-                       for key, value in report["circuits"].items()]
-    return not any("AUTO" not in status for status in circuits_status)
-
-
-def rack_and_circuit_report(site):
-    """Generate a report of the status of rack and circuit control.
-
-    :param str site: site id.
-    :return: Report with the status of compressors and circuits status.
-    :rtype: dict
-    """
-    report = {}
-    circuit_on_auto = is_circuit_control_in_auto(site)
-    compressors_on_auto = is_rack_control_in_auto(site)
-    if not isinstance(circuit_on_auto, bool) or not isinstance(compressors_on_auto, bool):
-        report["error"] = "Some results could not get obtained."
-        report["all_auto"] = False
-    else:
-        report["all_auto"] = True if circuit_on_auto and compressors_on_auto else False
-    report["all_circuit_on_auto"] = circuit_on_auto
-    report["all_compressors_on_auto"] = compressors_on_auto
-    return report
-
-
-def operate(args):
-    sites_to_operate = Operator.sites_to_operate(args.sites_to_operate, args.controller)
-    if not sites_to_operate:
-        return
-
-    print("\n\n--- Operation Report: ---")
-    sites_with_problems = []
-    sites_valid_circuits = []
-    sites_valid_compressors = []
+app = typer.Typer()
 
-    for site in sites_to_operate:
-        print(f"\n\nReport on site: {site}")
-        site_report = rack_and_circuit_report(site)
 
-        if site_report.pop("error", None):
-            sites_with_problems.append(site)
-            pprint(site_report)
-            continue
+class AutoReportManager:
+    def __init__(self):
+        self.sites_with_problems = []
+        self.sites_valid_circuits = []
+        self.sites_valid_compressors = []
+
+    @staticmethod
+    def generate_report(site):
+        """Generate a report of the status of rack and circuit control.
+
+        :return: Report with the status of compressors and circuits status.
+        :rtype: dict
+        """
+        def generate_data():
+            data = {"all_circuit_on_auto": circuits_on_auto,
+                    "all_compressors_on_auto": compressors_on_auto}
+            if not circuits_on_auto or not compressors_on_auto:
+                data.update({"error": "Some results could not get obtained.", "all_auto": False})
+            else:
+                all_auto = True if circuits_on_auto and compressors_on_auto else False
```

Comment:
An expression like 
```python
True if proposition else False
```
is equivalent to the proposition itself (or its boolean value to be more precise for Python)

 ```suggestion
                 all_auto = bool(circuits_on_auto and compressors_on_auto)
```